File: src/main/java/com/aquasecurity/plugins/trivy/model/Findings.java
package com.aquasecurity.plugins.trivy.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Findings {

    @JsonProperty("SchemaVersion")
    public int schemaVersion;
    @JsonProperty("ArtifactName")
    public String artifactName;
    @JsonProperty("ArtifactType")
    public String artifactType;
    @JsonProperty("Metadata")
    public Metadata metadata;
    @JsonProperty("Results")
    public List<Result> results;

    public List<Finding> getBySeverity(Severity severity) {
        return null;
    }

    public void setBasePath(String basePath) {
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/model/Findings.kt
package com.aquasecurity.plugins.trivy.model

import com.fasterxml.jackson.annotation.JsonProperty

class Findings {
    @JsonProperty("SchemaVersion")
    var schemaVersion: Int = 0

    @JsonProperty("ArtifactName")
    var artifactName: String? = null

    @JsonProperty("ArtifactType")
    var artifactType: String? = null

    @JsonProperty("Metadata")
    var metadata: Metadata? = null

    @JsonProperty("Results")
    var results: List<Result>? = null

    fun getBySeverity(severity: Severity?): List<Finding>? {
        return null
    }

    fun setBasePath(basePath: String?) {
    }
}